apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -128,6 +115,31 @@ where
         }
     }
 
+    /// Calculates what the [`TxResult`]s would be for the transactions in a
+    /// proposed block, as well as counting the number of digest transactions
+    /// present. `ProcessProposal` should be able to make a decision on whether
+    /// a proposed block is acceptable or not based solely on what this function
+    /// returns.
+    pub fn process_proposed_txs(
```

Comment:
Suggest we take the opportunity to rename this fn (and `process_single_tx`), since we are changing the signatures of these two fns anyway